core/domain/domain.go

```diff
@@ -0,0 +1,24 @@
+package domain
+
+import (
+	"context"
+	"time"
+)
+
+type Repository interface {
+	Create(ctx context.Context, domain Domain) error
+	Get(ctx context.Context, id string) (Domain, error)
+	Update(ctx context.Context, domain Domain) (Domain, error)
+	List(ctx context.Context, flt Filter) ([]Domain, error)
+	Delete(ctx context.Context, id string) error
+}
+
+type Domain struct {
+	ID         string
+	Name       string
+	OrgID      string
+	Token      string
+	Verified   bool
```

Comment:
Let's use a enum as `State` instead with possible values as `Pending` and `Verified`